AnkiHubSoftware/ankihub_addon
# [BUILD-486] feat: Move smart search button and change its look

On the Anki desktop add-on, Remove the “Select Flashcards” button and have the ‘Ankihub AI’ (robot icon) button do that function instead when selected on the Decks screen.

## Related issues
https://ankihub.atlassian.net/browse/BUILD-486


## Proposed changes
- Move code for showing the flashcard selector button and flashcard selector dialog from `deckbrowser.py` to overview.py`.
  - We will show the button on the deck overview screen now, instead of on the deck browser screen.
- Add code for new button (`add_flashcard_selector_button.js`), use it in `overview.py`.
- Call `overview.setup()` in `entrypoint.py`


## How to reproduce
- Make sure that you have the `show_flashcards_selector_button` feature flag active
- Change `ANKING_DECK_ID` in `settings.py` to the deck id of an AnkiHub deck you have installed
- Open Anki
- Click on the deck name
- You should see the flashcard selector button
- Hovering over it should show the tooltip


## Screenshots and videos
<img src="https://github.com/user-attachments/assets/14349c4a-7b45-40e4-bade-b6eeb5f93b03" width="700" />

<img src="https://github.com/user-attachments/assets/bf5f889b-507d-4dd5-ad56-1bd8e9f3b0e8" width="700" />


## Further comments
If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...
